tests/handlers/test_q30_handler.py

```diff
@@ -9,43 +9,79 @@ class TestQ30Handler(HandlerTestBase):
+        q30_handler.collect((key, value_6))
+        q30_handler.collect((key, value_7))
+        q30_handler.collect((key, value_8))
         self.q30_handler = q30_handler
 
     def set_qc_config(self, qc_config):
         self.q30_handler.qc_config = qc_config
 
     def test_all_is_fine(self):
-        qc_config = {'name': 'Q30Handler', 'error': 70, 'warning': 80}
+        qc_config = {"name": "Q30Handler", "error": 70, "warning": 80}
         self.set_qc_config(qc_config)
         errors_and_warnings = list(self.q30_handler.check_qc())
         self.assertIn("index read", f"{errors_and_warnings}")
-        self.assertEqual(f"{errors_and_warnings}", "[Fatal QC error: %Q30 50.00 was too low on lane: 1 for index read: 1]")
+        self.assertEqual(
+            f"{errors_and_warnings}",
+            "[Fatal QC error: %Q30 50.00 was too low on lane: 1 for index read: 1, Fatal QC error: %Q30 50.00 was too low on lane: 1 for index read: 2, Fatal QC error: %Q30 60.00 was too low on lane: 1 for read: 2, Fatal QC error: %Q30 50.00 was too low on lane: 2 for index read: 1, Fatal QC error: %Q30 50.00 was too low on lane: 2 for read: 2, Fatal QC error: %Q30 40.00 was too low on lane: 2 for read: 3]",
+        )
 
     def test_warning(self):
-        qc_config = {'name': 'Q30Handler', 'error': 70, 'warning': 85}
+        qc_config = {"name": "Q30Handler", "error": 70, "warning": 85}
         self.set_qc_config(qc_config)
         errors_and_warnings = list(self.q30_handler.check_qc())
-        self.assertEqual(len(errors_and_warnings), 2)
+        self.assertEqual(len(errors_and_warnings), 7)
 
         class_names = self.map_errors_and_warnings_to_class_names(errors_and_warnings)
-        self.assertListEqual(class_names, ['QCErrorWarning', 'QCErrorFatal'])
+        self.assertListEqual(
+            class_names,
+            [
+                "QCErrorWarning", "QCErrorFatal", "QCErrorFatal", "QCErrorFatal", "QCErrorFatal", "QCErrorFatal", "QCErrorFatal",
+            ],
+        )
 
     def test_error(self):
-        qc_config = {'name': 'Q30Handler', 'error': 95, 'warning': 98}
+        qc_config = {"name": "Q30Handler", "error": 95, "warning": 98}
         self.set_qc_config(qc_config)
         errors_and_warnings = list(self.q30_handler.check_qc())
-        self.assertEqual(len(errors_and_warnings), 3)
+        self.assertEqual(len(errors_and_warnings), 8)
 
         class_names = self.map_errors_and_warnings_to_class_names(errors_and_warnings)
-        self.assertListEqual(class_names, ['QCErrorFatal', 'QCErrorFatal', 'QCErrorFatal'])
+        self.assertListEqual(
+            class_names,
+            [
+                "QCErrorFatal", "QCErrorFatal", "QCErrorFatal", "QCErrorFatal", "QCErrorFatal", "QCErrorFatal", "QCErrorFatal", "QCErrorFatal",
+            ],
+        )
 
-if __name__ == '__main__':
+    def test_error_low_qc_marks(self):
+        qc_config = {"name": "Q30Handler", "error": 40, "warning": 60}
```

Comment:
I would suggest removing this test and add this config value to `test_warning` instead.